### ISSUE TITLE
Gradients under default dynamiqs tolerances can be off by 2x with no warning

### ISSUE THREAD

**Issue by ibralyousef** [quchip/quchip#51]
Found while piloting a five-device chip (two Duffing transmons, flux-tunable coupler, readout resonator, Purcell filter device with one port) on main + #50.

## Observation

`jax.grad` of a pulse observable with respect to `pulse.0.duration` through `QuantumSequence.simulate` on the dynamiqs backend:

| Case | Default options | `Dopri8(rtol=1e-10, atol=1e-12)` | Finite difference (h → 0) |
|---|---|---|---|
| 5-device chip, FluxDrive on coupler, d(overlap)/d(duration) | 0.320 | 0.1667 | 0.122, 0.153, 0.166 for h = 1, 0.25, 0.05 ns |
| 2-device q–c, ChargeDrive on c | −0.008180 | −0.008548 | −0.008548 |
| 2-device q–c, FluxDrive on c | 0.000737 | −0.02657 | 0.0003, −0.0114, −0.0258 (converging toward AD) |
| 1 transmon, ChargeDrive | −0.032995 | – | −0.03294 |

AD is correct once tolerances are tight; at the defaults it is silently wrong by a factor of two on a realistic chip and by 4.5 % on a two-device chip. Amplitude gradients on the 5-device chip were off by 1.5 % at defaults. Time-grid density (141 vs 1401 points) makes no difference.

## Why it matters

`docs/guides/choosing-a-backend.md` line 124 advises comparing at tighter tolerances, but nothing in the API enforces or flags it. A user following `examples/03_differentiate_a_driven_chip.md` with default options will fit or optimise against wrong derivatives.

## Suggestions (pick one)

- A documented gradient-safe preset, e.g. `options="gradient"` mapping to a tight adaptive method, used by the differentiability example.
- A one-time warning when `jax.grad` traces through `simulate` with default tolerances.
- At minimum, a prominent note in the differentiability guide with the numbers above.

## Reproducer

```python
import jax, jax.numpy as jnp, dynamiqs as dq
from quchip import Chip, DuffingTransmon, FluxTunableTransmon, Capacitive, ChargeDrive, QuantumSequence, GaussianEdge, RWA

def run(options):
    q = DuffingTransmon(freq=5.0, anharmonicity=-0.2, levels=3, label="q")
    c = FluxTunableTransmon(freq=6.5, anharmonicity=-0.2, levels=3, label="c")
    chip = Chip([q, c], [Capacitive(q, c, g=0.08)], frame="rotating", approximation=RWA(), backend="dynamiqs")
    chip.wire(ChargeDrive("c", label="d"))
    seq = QuantumSequence(chip)
    seq.schedule("d", envelope=GaussianEdge(duration=60.0, edge_duration=15.0, sigmas=3, amplitude=0.01), freq=6.5)
    ket = chip.state({"q": 1, "c": 0})
    tlist = jnp.linspace(0.0, 80.0, 81)
    def f(dur):
        res = seq.with_params({"pulse.0.duration": dur}).simulate(tlist=tlist, initial_state=ket, options=options)
        return jnp.abs(jnp.asarray(res.overlap(ket))[-1]) ** 2
    ad = float(jax.grad(f)(jnp.asarray(60.0)))
    fd = float((f(60.05) - f(59.95)) / 0.1)
    print(f"ad={ad:.6f} fd={fd:.6f}")

run(None)
run({"method": dq.method.Dopri8(rtol=1e-10, atol=1e-12, max_steps=1_000_000)})
```

Checkout: main 2706237 + PR #50. macOS, Python 3.11, dynamiqs from the pinned environment.
